### DISCUSSION TITLE
How can I mention belongs to many relation for the same table

### DISCUSSION THREAD

**Discussion by muthukumaran-m** [jenssegers/laravel-mongodb#2091] [General]
I have User table and connections table. I want to make a relationship which will return all the connections for that user.

connections table:

![image](https://user-images.githubusercontent.com/42160498/92219140-c8973500-eeb7-11ea-9541-91d98ba26a55.png)

The Auth::id() Maybe either in user_id or requested_user_id I want to get the connected user with the authorized user.

I have writen this but it gives no records. Am I missing something?

**Comment by zigzagdev** [comment]
@muthukumaran-m 

Hi !

This isn’t a single `belongsToMany` case because `connections` references `users` twice (`user_id `and `requested_user_id`).
Define two self-referencing relations (sent/received) and merge them.
So, please try following one .

```php
public function connectionsSent() {
  return $this->belongsToMany(User::class, 'connections', 'user_id', 'requested_user_id');
}

public function connectionsReceived() {
  return $this->belongsToMany(User::class, 'connections', 'requested_user_id', 'user_id');
}

public function connections() {
  return $this->connectionsSent()->get()->merge($this->connectionsReceived()->get());
}
```

Your `no records` issue is usually caused by checking only one side or swapping the pivot keys.
